README.md

```diff
@@ -90,40 +78,93 @@ path = "fake_lib.rs"
 [dependencies]
 log = "=0.3.6"
 
-[package.metadata.raze]
+[workspace.metadata.raze]
 # The path relative path to the Bazel workspace root (location of
 # WORKSPACE.bazel/WORKSPACE file). If no workspace file is found,
 # the current working directory is used.
 workspace_path = "//cargo"
 
-# The target to generate BUILD rules for.
-target = "x86_64-unknown-linux-gnu"
+# This causes aliases for dependencies to be rendered in the BUILD
+# file located next to this `Cargo.toml` file.
+package_aliases_dir = "."
+
+# The set of targets to generate BUILD rules for.
+targets = [
+    "x86_64-apple-darwin",
+    "x86_64-pc-windows-msvc",
+    "x86_64-unknown-linux-gnu",
+]
+
+# The two acceptable options are "Remote" and "Vendored" which
+# is used to idnicate whether the user is using a non-vendored or
+# vendored set of dependencies.
+genmode = "Remote"
 ```
 
-#### Generate buildable targets
+### Using existing Cargo.toml
 
-First, install the required tools for vendoring and generating BUILDable
-targets.
+Almost all canonical cargo setups should be able to function inplace with
+`cargo-raze`. Assuming the Cargo workspace is now nested under a Bazel workspace,
+Users can simply add [RazeSettings](./impl/src/settings.rs) to their Cargo.toml
+files to be used for generating Bazel files
 
-```bash
-$ cargo install cargo-raze
-```
+```toml
+# Above this line should be the contents of your Cargo.toml file
 
-Following that, vendor your dependencies from within the cargo/ directory. This
-will also update your `Cargo.lock` file.
+[package.metadata.raze]
+# The path relative path to the Bazel workspace root (location of
+# WORKSPACE.bazel/WORKSPACE file). If no workspace file is found,
+# the current working directory is used.
+workspace_path = "//cargo"
 
-```bash
-$ cargo vendor --versioned-dirs
+# This causes aliases for dependencies to be rendered in the BUILD
+# file located next to this `Cargo.toml` file.
+package_aliases_dir = "."
+
+# The set of targets to generate BUILD rules for.
+targets = [
+    "x86_64-apple-darwin",
+    "x86_64-pc-windows-msvc",
+    "x86_64-unknown-linux-gnu",
+]
+
+# The two acceptable options are "Remote" and "Vendored" which
+# is used to idnicate whether the user is using a non-vendored or
+# vendored set of dependencies.
+genmode = "Remote"
 ```
 
-Finally, generate your BUILD files, again from within the `cargo/` directory
+#### Cargo workspace projects
 
-```bash
-$ cargo raze
+In projects that use [cargo workspaces](cargo_workspaces) uses should organize
+all of their `raze` settings into the `[workspace.metadata.raze]` field in the
+top level `Cargo.toml` file which contains the `[workspace]` definition. These
+settings should be identical to the ones seen in `[package.metadata.raze]` in
+[the previous section](#Using existing Cargo.toml). However, crate settings may still
+be placed in the `Cargo.toml` files of the workspace memebers:
```

Comment:
Is this the reason for this "canonicalization" discussion from https://github.com/google/cargo-raze/pull/301?

Can we require all raze metadata live in the workspace root toml?

Reply:
I feel quite strongly that settings relevant to to dependencies for a specific package should be able to live in that package's `Cargo.toml`

Reply:
IIUC package in this context doesn't mean "free floating crate in the sea of crates", but rather "crate within a cargo workspace". What's the value add here that is worth the additional complexity imposed on the design?

Reply:
When I author the `Cargo.toml` file for my package (which your definition is correct in this context), and I add `serde-json` as a dependency, I shouldn't have to go over to my root `Cargo.toml` file to update it to be compatible with cargo raze. When working on a project, users should only need to deal with 1 manifest.

Reply:
I think you've convinced me of this, so for now, I'm approaching this in a different way in #301 where I"m basically hoping you can move the "nasty shenanigans"_(tm)_ out of settings processing and into CrateContext

Reply:
I think I was able to accomplish that. Right now these paths are relative to the cargo workspace root, not the current toml file (which might be bad? Certainly not ideal). I'm fine with it for now since the pattern seems to be that additional build files are generally put in `workspace_path` anyway. I think once the dust is settled there can be a small PR to improve that usability.

Reply:
Fully agreed with your assessment here. Marking resolved.